src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SNI/SSRP.cs

```diff
@@ -192,52 +191,77 @@ private static byte[] SendUDPRequest(string browserHostname, int port, byte[] re
 
                 IPAddress[] ipAddresses = SNICommon.GetDnsIpAddresses(browserHostname);
                 Debug.Assert(ipAddresses.Length > 0, "DNS should throw if zero addresses resolve");
-
+                IPAddress[] ipv4Addresses = null;
+                IPAddress[] ipv6Addresses = null;
                 switch (ipPreference)
                 {
                     case SqlConnectionIPAddressPreference.IPv4First:
                         {
-                            SsrpResult response4 = SendUDPRequest(ipAddresses.Where(i => i.AddressFamily == AddressFamily.InterNetwork).ToArray(), port, requestPacket, allIPsInParallel);
+                            SplitIPv4AndIPv6(ipv4Addresses, out ipv4Addresses, out ipv6Addresses);
```

Comment:
I agree.  I was wondering about that as well.  It turns out since this code is only run on Linux and requires sudo/administrative privileges to get the correct environment, there's currently no test coverage for this section unfortunately.